gateway/disk_package.py

```diff
@@ -0,0 +1,31 @@
+from pathlib import Path
+
+import gateway.deposit_directory
+from gateway.exceptions import NoContentError
+from gateway.package import Package
+
+class DiskPackage(Package):
+
+    def get_root_path(self) -> Path:
+        return self.deposit_dir.resolve(self.path)
+
+    def validate_path(self) -> None:
+        root_path = self.get_root_path()
+        if not root_path.exists():
+            raise NoContentError(f"No content exists at path {root_path}")
+
+    def __init__(self, deposit_dir: 'gateway.deposit_directory.DepositDirectory', path: Path):
```

Comment:
Can the `__init__` be the first method?

Reply:
Probably? I have a preference for defining methods before I use them, but you're not the only person who asked about this, so I'll try to change it to have the constructor first.

Reply:
Done. See https://github.com/mlibrary/dor-py/pull/15/commits/c4d94bd48e250c9bd8046fb9b42837b77036eae5